drivers/javascript/net.coffee

```diff
@@ -399,7 +399,7 @@ class TcpConnection extends Connection
         @rawSocket.on 'error', (args...) =>
             if @isOpen()
                 @close({noreplyWait:false})
-            @emit 'close'
+            @emit 'error', new err.RqlDriverError "Could not connect to " + @host + ":" + @port
```

Comment:
OK, thanks for being patient, I finally understand the driver code in depth enough to form an educated opinion on this PR.

I agree the `"error"` event should be raised again instead of `"close"`. But I think we should make three changes to the patch here:

1. Change the callback from taking `(args...)` to just `(err)`, since Node only passes an error object, not a bunch of args (this was an oversight in the existing code)
2. Instead of passing a new `RqlDriverError` as the argument to `emit`, just pass `err`. In the Connection superclass, it registers a listener for `"error"` that will raise this same RqlDriverError, so it's redundant here.
3. Remove the `isOpen()` check in here, since (as you mention) the `"close"` event gets emitted immediately afterwards, it's redundant.

The final code should just be like:

```coffee
@rawSocket.on('error', (err) => @emit('error', err))
```